### ISSUE TITLE
[QUESTION] Why does GRGlInterface.CreateDefaultInterface(); always return null

### ISSUE THREAD

**Issue by junioraepi** [mono/SkiaSharp#828]
I am trying to write code that resizes an image using Skiasharp using an accelerated SKSurface.   Unfortunately I am getting a null back when I call ```GRGlInterface.CreateDefaultInterface();``` on Windows (I have not checked in any other OS yet).  What am I doing wrong?

```C#
public Stream ResizeImage(Stream inputStream, int width, int height)
        {
            // This statement always returns null
            var glInterface = GRGlInterface.CreateDefaultInterface();
            var context = GRContext.Create(GRBackend.OpenGL, glInterface);

            var glInfo = new GRGlFramebufferInfo(
                fboId: 0,
                format: SKColorType.Rgba8888.ToGlSizedFormat());
            var renderTarget = new GRBackendRenderTarget(
                width: width,
                height: height,
                sampleCount: 0,
                stencilBits: 0,
                glInfo: glInfo);
            var surface = SKSurface.Create(
                context, renderTarget, GRSurfaceOrigin.BottomLeft, SKColorType.Rgba8888);

            var canvas = surface.Canvas;

            // Read the original image
            using (var stream = new SKManagedStream(inputStream))
            using (var bitmap = SKBitmap.Decode(stream))
            {
                // Draw the image
                using (var paint = new SKPaint
                {
                    FilterQuality = SKFilterQuality.Low
                })
                {
                    canvas.DrawColor(SKColor.Parse("00ff00"));
                    canvas.DrawBitmap(bitmap, SKRect.Create(width, height), paint);
                }

                // Render to a png
                SKImage image = surface.Snapshot().ToRasterImage();
                SKData encoded = image.Encode(SKEncodedImageFormat.Png, 80);
                Stream outputStream = encoded.AsStream();
                return outputStream;
            }
        }
```

**Comment by mattleibow** [comment]
That might be because you don't have an OpenGL context set up?

SkiaSharp does not create an OpenGL context, the "interface" here is basically a set of function pointers to the OpenGL functions (glDrawXXX). In my SkiaSharp.Views, I first set up a context to draw to the control/memory and then in the draw loop, I get the interface.

For the Windows Desktop, I use OpenTK and GLControl (I will be switching eventually to DirectX): https://github.com/mono/SkiaSharp/blob/master/source/SkiaSharp.Views/SkiaSharp.Views.Desktop/SKGLControl.cs

For UWP, I set up my ANGLE/DX context:
https://github.com/mono/SkiaSharp/blob/master/source/SkiaSharp.Views/SkiaSharp.Views.UWP/SKSwapChainPanel.cs

https://github.com/mono/SkiaSharp/issues/213#issuecomment-272755110
